tool/tsh/common/tsh.go

```diff
@@ -3703,6 +3721,31 @@ func onResolve(cf *CLIConf) error {
 
 // onSSH executes 'tsh ssh' command
 func onSSH(cf *CLIConf) error {
+	// If "tsh ssh -V" is invoked, tsh is in OpenSSH compatibility mode, show
+	// the version and exit.
+	if cf.ShowVersion {
+		modules.GetModules().PrintVersion()
+		return nil
+	}
+
+	// If "tsh ssh" is invoked with the "-t" or "-T" flag, manually validate
+	// "-t" and "-T" flags for "tsh ssh" due to the lack of inverse short flags
+	// in kingpin.
+	if cf.Interactive && cf.NonInteractive {
```

Comment:
@rosstimothy That's what I originally did and even had a working implementation.

I don't remember all the details, but during testing I found that the way we hook kingpin for terminate (which I would have to use) is broken and the fact that `tsh` does not panic often is incidental. Fixing this would have been a fairly big undertaking because we hook termination from `tsh`.

This is what I have in my notes:

```
https://github.com/gravitational/kingpin/blob/master/app.go#L617

None of those `a.terminate(1)` will actually exit due to how we hook the terminate function
in Teleport, which means things will not be set downstream and panic when it flows through
the rest of kingpin code.

This FatalIfError is called all over the place in kingpin.

https://github.com/gravitational/kingpin/blob/master/app.go#L642C23-L642C35
```